src/balance/Balance.chain.ts

```diff
@@ -75,17 +72,23 @@ export async function listenToBalanceChanges(
     balance: BN,
     change: BN
   ) => void
-): Promise<UnsubscribeHandle> {
+): Promise<BN> {
   const blockchain = await getCached()
-  let previous = await getBalance(accountAddress)
-  return blockchain.api.query.balances.freeBalance<Balance>(
-    accountAddress,
-    (current: Balance) => {
-      const change = current.sub(previous)
-      previous = current
-      listener(accountAddress, current, change)
-    }
-  )
+  let {
+    data: { free: previousFree },
+  } = await blockchain.api.query.system.account(accountAddress)
+
+  if (listener) {
+    blockchain.api.query.system.account(
+      accountAddress,
+      ({ data: { free: currentFree } }) => {
+        const change = currentFree.sub(previousFree)
+        previousFree = currentFree
+        listener(accountAddress, currentFree, change)
+      }
+    )
+  }
+  return previousFree
```

Comment:
Probably a merge thing. We need the unsubscribe handle here. Could you restore the previous implementation?

Reply:
Fixed in b1c22bd. Note: `Promise<UnsubscribeHandle === UnsubscribePromise` from Polkadot.js